session10/src/Homework/Ex7.java

```diff
@@ -0,0 +1,24 @@
+package Homework;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+
+public class Ex7 {
+
+    public static void main(String[] args) {
+
+        ArrayList<String> fruits = new ArrayList<>(Arrays.asList("mere", "pere", "banane"));
+        System.out.println("Unsorted arraylist: " + fruits);
+        Collections.sort(fruits);
+        System.out.println("Sorted arraylist: " + fruits);
+        Collections.sort(fruits, Collections.reverseOrder());
+        Collections.sort(fruits, Comparator.comparing(String::length));
```

Comment:
hmm, these comparators doesn't seem to act as it should. The final result I get is: 
`Sorted arraylist in descending order based on length: [pere, mere, banane]`. 
This is because you are sorting the elements in the reverse order (of their letters) and then you sort them ascending by their length.
Instead it should be something like (chained):
```
fruits.sort(Comparator.comparing(String::length).reversed()
                .thenComparing(Comparator.naturalOrder())); // for the sake of the demo we put this, but actually is not needed because by default they will be sorted by natural order (alphabetically for strings)
```
However, interesting fact that you used this expression `String::length`. Can you tell what it means? :)